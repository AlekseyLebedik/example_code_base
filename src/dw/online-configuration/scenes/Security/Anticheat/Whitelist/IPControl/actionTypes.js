export const IP_CONTROL_SEARCH = 'security/IP_CONTROL_SEARCH';

export const IP_CONTROL_FETCH = 'security/IP_CONTROL_FETCH';
export const IP_CONTROL_FETCH_SUCCESS = 'security/IP_CONTROL_FETCH_SUCCESS';

export const IP_CONTROL_FETCH_ALL = 'security/IP_CONTROL_FETCH_ALL';
export const IP_CONTROL_FETCH_ALL_SUCCESS =
  'security/IP_CONTROL_FETCH_ALL_SUCCESS';

export const IP_CONTROL_FETCH_GROUPS = 'security/IP_CONTROL_FETCH_GROUPS';
export const IP_CONTROL_FETCH_NOTES = 'security/IP_CONTROL_FETCH_NOTES';
export const IP_CONTROL_FETCH_GROUPS_SUCCESS =
  'security/IP_CONTROL_FETCH_GROUPS_SUCCESS';
export const IP_CONTROL_FETCH_NOTES_SUCCESS =
  'security/IP_CONTROL_FETCH_NOTES_SUCCESS';

export const IP_CONTROL_ADD_NOTE = 'security/IP_CONTROL_NOTE_ADD';
export const IP_CONTROL_ADD_NOTE_SUCCESS =
  'security/IP_CONTROL_ADD_NOTE_SUCCESS';
export const IP_CONTROL_ADD_NOTE_FAILED = 'security/IP_CONTROL_ADD_NOTE_FAILED';

export const IP_CONTROL_UPDATE_NOTE = 'security/IP_CONTROL_NOTE_UPDATE';
export const IP_CONTROL_UPDATE_NOTE_SUCCESS =
  'security/IP_CONTROL_UPDATE_NOTE_SUCCESS';
export const IP_CONTROL_UPDATE_NOTE_FAILED =
  'security/IP_CONTROL_UPDATE_NOTE_FAILED';

export const IP_CONTROL_DELETE_NOTE = 'security/IP_CONTROL_NOTE_DELETE';
export const IP_CONTROL_DELETE_NOTE_SUCCESS =
  'security/IP_CONTROL_DELETE_NOTE_SUCCESS';
export const IP_CONTROL_DELETE_NOTE_FAILED =
  'security/IP_CONTROL_DELETE_NOTE_FAILED';

export const IP_CONTROL_ADD_GROUP = 'security/IP_CONTROL_GROUP_ADD';
export const IP_CONTROL_ADD_GROUP_SUCCESS =
  'security/IP_CONTROL_ADD_GROUP_SUCCESS';
export const IP_CONTROL_ADD_GROUP_FAILED =
  'security/IP_CONTROL_ADD_GROUP_FAILED';

export const IP_CONTROL_DELETE_GROUP = 'security/IP_CONTROL_GROUP_DELETE';
export const IP_CONTROL_DELETE_GROUP_SUCCESS =
  'security/IP_CONTROL_DELETE_GROUP_SUCCESS';
export const IP_CONTROL_DELETE_GROUP_FAILED =
  'security/IP_CONTROL_DELETE_GROUP_FAILED';

export const IP_CONTROL_ADD_MODAL_OPEN = 'security/IP_CONTROL_ADD_MODAL_OPEN';
export const IP_CONTROL_ADD_MODAL_CLOSE = 'security/IP_CONTROL_ADD_MODAL_CLOSE';
export const IP_CONTROL_ADD = 'security/IP_CONTROL_ADD';
export const IP_CONTROL_ADD_SUCCESS = 'security/IP_CONTROL_ADD_SUCCESS';
export const IP_CONTROL_ADD_FAILED = 'security/IP_CONTROL_ADD_FAILED';

export const IP_CONTROL_DELETE = 'security/IP_CONTROL_DELETE';
export const IP_CONTROL_DELETE_SUCCESS = 'security/IP_CONTROL_DELETE_SUCCESS';

export const WHITELISTED_USERS_FETCH = 'security/WHITELISTED_USERS_FETCH';
export const WHITELISTED_USERS_FETCH_SUCCESS =
  'security/WHITELISTED_USERS_FETCH_SUCCESS';
