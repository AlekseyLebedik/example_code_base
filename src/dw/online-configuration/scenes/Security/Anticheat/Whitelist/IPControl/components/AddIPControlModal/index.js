import AddIPControlModal from './container';

export default AddIPControlModal;
