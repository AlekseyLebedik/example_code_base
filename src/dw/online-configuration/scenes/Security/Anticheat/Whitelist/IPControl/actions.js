import * as GlobalSnackBarActions from '@demonware/devzone-core/components/GlobalSnackBar/actions';
import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import { CriticalErrorActions } from 'dw/core/components/CriticalError';

import * as AT from './actionTypes';

export const onSearch = q => ({
  type: AT.IP_CONTROL_SEARCH,
  q,
});

export function fetchIPControl(params, append = false) {
  return {
    type: AT.IP_CONTROL_FETCH,
    params,
    append,
  };
}

export function fetchIPControlSuccess(payload, append) {
  return {
    type: AT.IP_CONTROL_FETCH_SUCCESS,
    payload,
    append,
  };
}

export function fetchIPControlFailed(err, action) {
  return dispatch => {
    dispatch(CriticalErrorActions.show(err, () => action));
  };
}

export function fetchAllIPControl(params, append = false) {
  return {
    type: AT.IP_CONTROL_FETCH_ALL,
    params,
    append,
  };
}

export function fetchAllIPControlSuccess(payload, append) {
  return {
    type: AT.IP_CONTROL_FETCH_ALL_SUCCESS,
    payload,
    append,
  };
}

export function fetchAllIPControlFailed(err, action) {
  return dispatch => {
    dispatch(CriticalErrorActions.show(err, () => action));
  };
}

export function fetchIPGroups(params, append) {
  return {
    type: AT.IP_CONTROL_FETCH_GROUPS,
    params,
    append,
  };
}

export function fetchIPGroupsSuccess(payload, append) {
  return {
    type: AT.IP_CONTROL_FETCH_GROUPS_SUCCESS,
    payload,
    append,
  };
}

export function fetchIPGroupsFailed(err, action) {
  return dispatch => {
    dispatch(CriticalErrorActions.show(err, () => action));
  };
}

export function fetchIPNotes(params, append) {
  return {
    type: AT.IP_CONTROL_FETCH_NOTES,
    params,
    append,
  };
}

export function fetchIPNotesSuccess(payload, append) {
  return {
    type: AT.IP_CONTROL_FETCH_NOTES_SUCCESS,
    payload,
    append,
  };
}

export function fetchIPNotesFailed(err, action) {
  return dispatch => {
    dispatch(CriticalErrorActions.show(err, () => action));
  };
}

export const addIPNote = values => ({
  type: AT.IP_CONTROL_ADD_NOTE,
  values,
});

export const updateIPNote = values => ({
  type: AT.IP_CONTROL_UPDATE_NOTE,
  values,
});

export const deleteIPNote = values => ({
  type: AT.IP_CONTROL_DELETE_NOTE,
  values,
});

export const addIPNoteSuccess = values => dispatch => {
  dispatch({
    type: AT.IP_CONTROL_ADD_NOTE_SUCCESS,
    payload: values,
  });
  dispatch(GlobalSnackBarActions.show('IP Note added.', 'success'));
};

export const updateIPNoteSuccess = values => dispatch => {
  dispatch({
    type: AT.IP_CONTROL_UPDATE_NOTE_SUCCESS,
    payload: values,
  });
  dispatch(GlobalSnackBarActions.show('IP Note updated.', 'success'));
};

export const deleteIPNoteSuccess = values => dispatch => {
  dispatch({
    type: AT.IP_CONTROL_DELETE_NOTE_SUCCESS,
    payload: values,
  });
  dispatch(GlobalSnackBarActions.show('IP Note deleted.', 'success'));
};

export const addIPNoteFailed = err => dispatch => {
  dispatch(nonCriticalHTTPError(err));
  dispatch({
    type: AT.IP_CONTROL_ADD_NOTE_FAILED,
  });
};

export const addIPGroupFailed = err => dispatch => {
  dispatch(nonCriticalHTTPError(err));
  dispatch({
    type: AT.IP_CONTROL_ADD_GROUP_FAILED,
  });
};

export const updateIPNoteFailed = err => dispatch => {
  dispatch(nonCriticalHTTPError(err));
  dispatch({
    type: AT.IP_CONTROL_UPDATE_NOTE_FAILED,
  });
};

export const deleteIPNoteFailed = err => dispatch => {
  dispatch(nonCriticalHTTPError(err));
  dispatch({
    type: AT.IP_CONTROL_DELETE_NOTE_FAILED,
  });
};

export const deleteIPGroupFailed = err => dispatch => {
  dispatch(nonCriticalHTTPError(err));
  dispatch({
    type: AT.IP_CONTROL_DELETE_GROUP_FAILED,
  });
};

export const addIPGroup = values => ({
  type: AT.IP_CONTROL_ADD_GROUP,
  values,
});

export const addIPGroupSuccess = values => dispatch => {
  dispatch({
    type: AT.IP_CONTROL_ADD_GROUP_SUCCESS,
    payload: values,
  });
  dispatch(
    GlobalSnackBarActions.show('IP Group added successfully.', 'success')
  );
};

export const deleteIPGroup = values => ({
  type: AT.IP_CONTROL_DELETE_GROUP,
  values,
});

export const deleteIPGroupSuccess = values => dispatch => {
  dispatch({
    type: AT.IP_CONTROL_DELETE_GROUP_SUCCESS,
    payload: values,
  });
  dispatch(
    GlobalSnackBarActions.show('IP Group deleted successfully.', 'success')
  );
};

export const openAddModal = () => ({
  type: AT.IP_CONTROL_ADD_MODAL_OPEN,
});

export const closeAddModal = () => ({
  type: AT.IP_CONTROL_ADD_MODAL_CLOSE,
});

export const addIPControl = values => ({
  type: AT.IP_CONTROL_ADD,
  values,
});

export const addIPControlSuccess = values => dispatch => {
  dispatch({
    type: AT.IP_CONTROL_ADD_SUCCESS,
    payload: values,
  });
  dispatch(
    GlobalSnackBarActions.show('IP/Range added successfully.', 'success')
  );
  dispatch(fetchAllIPControl());
};

export const addIPControlFailed = err => dispatch => {
  dispatch(nonCriticalHTTPError(err));
  dispatch({
    type: AT.IP_CONTROL_ADD_FAILED,
  });
};

export const deleteIPControl = (ips, groups, checkedOption) => dispatch => {
  dispatch({
    type: AT.IP_CONTROL_DELETE,
    ips,
    groups,
    checkedOption,
  });
  dispatch(
    GlobalSnackBarActions.show('IP/Ranges deleted successfully.', 'success')
  );
  if (checkedOption) {
    groups.forEach(g => {
      dispatch({
        type: AT.IP_CONTROL_DELETE_GROUP,
        id: g,
        deleteContents: false,
      });
    });
  }
};

export const deleteIPControlSuccess = ips => ({
  type: AT.IP_CONTROL_DELETE_SUCCESS,
  ips,
});

export const deleteIPControlFailed = err => dispatch => {
  dispatch(nonCriticalHTTPError(err));
};

export function fetchWhitelistedUsers(params, append = false) {
  return {
    type: AT.WHITELISTED_USERS_FETCH,
    params,
    append,
  };
}

export function fetchWhitelistedUsersSuccess(payload, append) {
  return {
    type: AT.WHITELISTED_USERS_FETCH_SUCCESS,
    payload,
    append,
  };
}

export function fetchWhitelistedUsersFailed(err, action) {
  return dispatch => {
    dispatch(CriticalErrorActions.show(err, () => action));
  };
}
