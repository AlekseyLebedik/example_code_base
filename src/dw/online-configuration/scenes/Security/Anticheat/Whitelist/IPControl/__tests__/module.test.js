import { CriticalErrorActions } from 'dw/core/components/CriticalError';
import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import * as GlobalSnackBarActions from '@demonware/devzone-core/components/GlobalSnackBar/actions';
import ModalHandlers from 'dw/core/components/ModalHandlers';
import * as actions from '../actions';
import * as AT from '../actionTypes';
import reducer from '../reducer';
import * as selectors from '../selectors';
import * as helper from '../helpers';

jest.mock('@demonware/devzone-core/components/GlobalSnackBar/actions');
jest.mock('dw/core/components/ModalHandlers');
jest.mock('dw/core/components/CriticalError');
jest.mock('@demonware/devzone-core/helpers/errors');

const ipId = 292531;
const q = 'blah';
const ip = {
  group: 'Unassigned Group',
  groupID: 0,
  ipInt: ipId,
  ipAddr: '0.4.118.179',
  ipRange: 41,
  creatorNote: '',
  note: 'Click to add note',
  type: 'IP/Ranges',
  updatedAt: 'N/A',
};
const nextPageToken = 'CAE';
const data = [
  ip,
  {
    group: 'Unassigned Group',
    groupID: 0,
    ipInt: 5285094,
    ipAddr: '0.80.164.230',
    ipRange: 32,
    creatorNote: '',
    note: 'Click to add note',
    type: 'IP/Ranges',
    updatedAt: 'N/A',
  },
];
const payload = { data, nextPageToken };
const values = [
  {
    ipAddr: '127.0.0.1',
    ipRange: 5,
    ipInt: 2130706433,
  },
];
const err = 'error';

const userA = {
  userId: 'user_a',
  group: 'Unassigned Group',
  groupID: 0,
  creatorNote: '',
  note: 'Click to add note',
  type: 'Player UserID',
  updatedAt: 'N/A',
};
const usersData = [userA];

describe('Security - Anticheat - Whitelist - IPControl', () => {
  describe('Action Creators', () => {
    const MOCKED_CRITICAL_ERROR = 'MOCKED_CRITICAL_ERROR';
    const MOCKED_NON_CRITICAL_ERROR = 'MOCKED_NON_CRITICAL_ERROR';
    const MOCKED_GLOBAL_SNACK_BAR = 'MOCKED_GLOBAL_SNACK_BAR_SHOW';
    const MOCKED_MODAL_HANDLERS_CLOSE = 'MOCKED_MODAL_HANDLERS_CLOSE';
    const MOCKED_MODAL_HANDLERS_SUBMIT = 'MOCKED_MODAL_HANDLERS_SUBMIT';
    const MOCKED_MODAL_HANDLERS_STOP_SUBMITTING =
      'MOCKED_MODAL_HANDLERS_STOP_SUBMITTING';

    beforeAll(() => {
      CriticalErrorActions.show.mockReturnValue({
        type: MOCKED_CRITICAL_ERROR,
      });
      GlobalSnackBarActions.show.mockReturnValue({
        type: MOCKED_GLOBAL_SNACK_BAR,
      });
      ModalHandlers.close.mockReturnValue({
        type: MOCKED_MODAL_HANDLERS_CLOSE,
      });
      ModalHandlers.submit.mockReturnValue({
        type: MOCKED_MODAL_HANDLERS_SUBMIT,
      });
      ModalHandlers.stopSubmitting.mockReturnValue({
        type: MOCKED_MODAL_HANDLERS_STOP_SUBMITTING,
      });
      nonCriticalHTTPError.mockReturnValue({
        type: MOCKED_NON_CRITICAL_ERROR,
      });
    });

    it('IP_CONTROL_SEARCH', () => {
      const action = actions.onSearch(q);

      expect(action).toHaveProperty('type', AT.IP_CONTROL_SEARCH);
      expect(action).toHaveProperty('q', q);
    });

    it('IP_CONTROL_FETCH', () => {
      const action = actions.fetchIPControl({});

      expect(action).toHaveProperty('type', AT.IP_CONTROL_FETCH);
      expect(action).toHaveProperty('params', {});
      expect(action).toHaveProperty('append', false);
    });

    it('IP_CONTROL_FETCH_SUCCESS', () => {
      const action = actions.fetchIPControlSuccess(payload, true);

      expect(action).toHaveProperty('type', AT.IP_CONTROL_FETCH_SUCCESS);
      expect(action).toHaveProperty('payload', payload);
      expect(action).toHaveProperty('append', true);
    });

    it('fetchIPControlFailed', () => {
      const action = actions.fetchIPControlFailed();

      expect(action).toAsyncDispatch({ type: MOCKED_CRITICAL_ERROR });
    });

    it('WHITELISTED_USERS_FETCH', () => {
      const action = actions.fetchWhitelistedUsers({});

      expect(action).toHaveProperty('type', AT.WHITELISTED_USERS_FETCH);
      expect(action).toHaveProperty('params', {});
      expect(action).toHaveProperty('append', false);
    });

    it('WHITELISTED_USERS_FETCH_SUCCESS', () => {
      const action = actions.fetchWhitelistedUsersSuccess(payload, true);

      expect(action).toHaveProperty('type', AT.WHITELISTED_USERS_FETCH_SUCCESS);
      expect(action).toHaveProperty('payload', payload);
      expect(action).toHaveProperty('append', true);
    });

    it('fetchWhitelistedUsersFailed', () => {
      const action = actions.fetchWhitelistedUsersFailed();

      expect(action).toAsyncDispatch({ type: MOCKED_CRITICAL_ERROR });
    });

    it('IP_CONTROL_ADD_MODAL_OPEN', () => {
      expect(actions.openAddModal()).toHaveProperty(
        'type',
        AT.IP_CONTROL_ADD_MODAL_OPEN
      );
    });

    it('IP_CONTROL_ADD_MODAL_CLOSE', () => {
      expect(actions.closeAddModal()).toHaveProperty(
        'type',
        AT.IP_CONTROL_ADD_MODAL_CLOSE
      );
    });

    it('IP_CONTROL_ADD', () => {
      const action = actions.addIPControl(values);

      expect(action).toHaveProperty('type', AT.IP_CONTROL_ADD);
      expect(action).toHaveProperty('values', values);
    });

    it('IP_CONTROL_ADD_SUCCESS', () => {
      const action = actions.addIPControlSuccess(values);

      expect(action).toAsyncDispatch(
        { type: AT.IP_CONTROL_ADD_SUCCESS, payload: values },
        { type: MOCKED_GLOBAL_SNACK_BAR }
      );
    });

    it('addIPControlFailed', () => {
      const action = actions.addIPControlFailed(err);

      expect(action).toAsyncDispatch(
        { type: MOCKED_NON_CRITICAL_ERROR },
        { type: AT.IP_CONTROL_ADD_FAILED }
      );
    });
    // IP GROUPS start
    it('fetchIPGroups action', () => {
      const action = actions.fetchIPGroups('test', 'test');
      expect(action).toMatchSnapshot();
    });

    it('fetchIPGroupsSuccess action', () => {
      const action = actions.fetchIPGroupsSuccess('test', 'test');
      expect(action).toMatchSnapshot();
    });

    it('fetchIPGroupsFailed action', () => {
      CriticalErrorActions.show.mockClear();
      const mockDispatch = jest.fn();
      actions.fetchIPGroupsFailed()(mockDispatch);
      expect(mockDispatch).toHaveBeenCalled();
      expect(CriticalErrorActions.show).toHaveBeenCalled();
    });
    // IP GROUPS end
    // IP NOTES start
    it('fetchIPNotes action', () => {
      const action = actions.fetchIPNotes('test', 'test');
      expect(action).toMatchSnapshot();
    });

    it('fetchIPNotesSuccess action', () => {
      const action = actions.fetchIPNotesSuccess('test', 'test');
      expect(action).toMatchSnapshot();
    });

    it('fetchIPNotesFailed action', () => {
      CriticalErrorActions.show.mockClear();
      const mockDispatch = jest.fn();
      actions.fetchIPNotesFailed()(mockDispatch);
      expect(mockDispatch).toHaveBeenCalled();
      expect(CriticalErrorActions.show).toHaveBeenCalled();
    });

    it('addIPNote action', () => {
      const action = actions.addIPNote('test');
      expect(action).toMatchSnapshot();
    });

    it('updateIPNote action', () => {
      const action = actions.updateIPNote('test');
      expect(action).toMatchSnapshot();
    });

    it('deleteIPNote action', () => {
      const action = actions.deleteIPNote('test');
      expect(action).toMatchSnapshot();
    });

    it('addIPNoteSuccess action', () => {
      GlobalSnackBarActions.show.mockClear();
      const mockDispatch = jest.fn();
      actions.addIPNoteSuccess('test')(mockDispatch);
      expect(mockDispatch).toHaveBeenCalled();
      expect(GlobalSnackBarActions.show).toHaveBeenCalled();
    });

    it('updateIPNoteSuccess action', () => {
      GlobalSnackBarActions.show.mockClear();
      const mockDispatch = jest.fn();
      actions.updateIPNoteSuccess('test')(mockDispatch);
      expect(mockDispatch).toHaveBeenCalled();
      expect(GlobalSnackBarActions.show).toHaveBeenCalled();
    });

    it('deleteIPNoteSuccess action', () => {
      GlobalSnackBarActions.show.mockClear();
      const mockDispatch = jest.fn();
      actions.deleteIPNoteSuccess('test')(mockDispatch);
      expect(mockDispatch).toHaveBeenCalled();
      expect(GlobalSnackBarActions.show).toHaveBeenCalled();
    });

    it('addIPNoteFailed action', () => {
      const mockDispatch = jest.fn();
      actions.addIPNoteFailed('test')(mockDispatch);
      expect(mockDispatch).toHaveBeenCalledWith({
        type: AT.IP_CONTROL_ADD_NOTE_FAILED,
      });
    });

    it('addIPGroupFailed action', () => {
      const mockDispatch = jest.fn();
      actions.addIPGroupFailed('test')(mockDispatch);
      expect(mockDispatch).toHaveBeenCalledWith({
        type: AT.IP_CONTROL_ADD_GROUP_FAILED,
      });
    });

    it('updateIPNoteFailed action', () => {
      const mockDispatch = jest.fn();
      actions.updateIPNoteFailed('test')(mockDispatch);
      expect(mockDispatch).toHaveBeenCalledWith({
        type: AT.IP_CONTROL_UPDATE_NOTE_FAILED,
      });
    });

    it('deleteIPGroupFailed action', () => {
      const mockDispatch = jest.fn();
      actions.deleteIPGroupFailed('test')(mockDispatch);
      expect(mockDispatch).toHaveBeenCalledWith({
        type: AT.IP_CONTROL_DELETE_GROUP_FAILED,
      });
    });

    // IP NOTES end

    it('IP_CONTROL_DELETE', () => {
      const action = actions.deleteIPControl([ipId]);

      expect(action).toAsyncDispatch(
        { type: AT.IP_CONTROL_DELETE, ips: [ipId] },
        { type: MOCKED_GLOBAL_SNACK_BAR }
      );
    });

    it('IP_CONTROL_DELETE_SUCCESS', () => {
      const action = actions.deleteIPControlSuccess([ipId]);

      expect(action).toHaveProperty('type', AT.IP_CONTROL_DELETE_SUCCESS);
      expect(action).toHaveProperty('ips', [ipId]);
    });

    it('deleteIPControlFailed', () => {
      const action = actions.deleteIPControlFailed(err);

      expect(action).toAsyncDispatch({ type: MOCKED_NON_CRITICAL_ERROR });
    });
  });

  describe('Reducer', () => {
    it('return the initial state', () => {
      expect(reducer(undefined, {})).toMatchSnapshot();
    });

    it('handle IP_CONTROL_SEARCH', () => {
      expect(reducer(undefined, actions.onSearch(q))).toMatchSnapshot();
    });

    it('handle IP_CONTROL_FETCH_SUCCESS: empty results', () => {
      expect(
        reducer(undefined, actions.fetchIPControlSuccess({ data: [] }))
      ).toMatchSnapshot();
    });

    it('handle IP_CONTROL_FETCH_SUCCESS: ipControl and nextPagetoken assignation', () => {
      expect(
        reducer(undefined, actions.fetchIPControlSuccess(payload))
      ).toMatchSnapshot();
    });

    it('Test append = false', () => {
      expect(
        reducer(
          { ipControl: [{ ipInt: 1 }, { ipInt: 2 }] },
          actions.fetchIPControlSuccess(
            { data: [{ ipInt: 3 }, { ipInt: 4 }] },
            false
          )
        )
      ).toMatchSnapshot();
    });

    it('Test append = true', () => {
      expect(
        reducer(
          { ipControl: [{ ipInt: 1 }, { ipInt: 2 }] },
          actions.fetchIPControlSuccess(
            { data: [{ ipInt: 3 }, { ipInt: 4 }] },
            true
          )
        )
      ).toMatchSnapshot();
    });

    it('handle IP_CONTROL_ADD_MODAL_OPEN', () => {
      expect(reducer(undefined, actions.openAddModal())).toMatchSnapshot();
    });

    it('handle IP_CONTROL_ADD_MODAL_CLOSE', () => {
      expect(reducer(undefined, actions.closeAddModal())).toMatchSnapshot();
    });

    it('handle IP_CONTROL_ADD', () => {
      expect(
        reducer(undefined, actions.addIPControl(values))
      ).toMatchSnapshot();
    });

    it('handle IP_CONTROL_ADD_SUCCESS', () => {
      expect(
        reducer(undefined, { type: AT.IP_CONTROL_ADD_SUCCESS, payload: values })
      ).toMatchSnapshot();
    });

    it('handle IP_CONTROL_ADD_FAILED', () => {
      expect(
        reducer(undefined, { type: AT.IP_CONTROL_ADD_FAILED })
      ).toMatchSnapshot();
    });

    it('handle IP_CONTROL_DELETE_SUCCESS', () => {
      const state = reducer(
        undefined,
        actions.fetchIPControlSuccess(payload, false)
      );
      expect(
        reducer(state, actions.deleteIPControlSuccess([ipId]))
      ).toMatchSnapshot();
    });
  });

  describe('Selectors', () => {
    it('validate searchParams', () => {
      const state = {
        Scenes: {
          Security: { Anticheat: { Whitelist: { IPControl: { q } } } },
        },
      };

      expect(selectors.searchParams(state)).toBe(q);
    });

    it('validate filteredSelector', () => {
      const state = {
        user: {
          profile: { timezone: null },
        },
        Scenes: {
          Security: {
            Anticheat: {
              Whitelist: {
                IPControl: {
                  ipControl: data,
                  q: { default: true, q: '0.4.118.179' },
                },
              },
            },
          },
        },
      };

      expect(selectors.filteredSelector(state)).toEqual([ip]);
    });

    it('validate filteredSelector, empty search', () => {
      const state = {
        user: {
          profile: { timezone: null },
        },
        Scenes: {
          Security: {
            Anticheat: { Whitelist: { IPControl: { ipControl: data } } },
          },
        },
      };

      expect(selectors.filteredSelector(state)).toEqual(data);
    });

    it('validate usersGroupAndDateSelector', () => {
      const state = {
        user: {
          profile: { timezone: null },
        },
        Scenes: {
          Security: {
            Anticheat: {
              Whitelist: {
                IPControl: {
                  whitelistedUsers: usersData,
                  q: { default: true, q: 'user_a' },
                },
              },
            },
          },
        },
      };

      expect(selectors.usersGroupAndDateSelector(state)).toEqual([userA]);
    });
  });
});

describe('Security - Anticheat - Whitelist - Helpers', () => {
  it('validates mapping whitelist type', () => {
    const common = {
      group: 'Unassigned Group',
      groupID: 0,
      creatorNote: '',
      note: 'Click to add note',
      type: 'IP/Ranges',
      updatedAt: 'N/A',
    };
    expect(
      helper.mapWhitelistType({
        ...common,
        ipInt: ipId,
        ipAddr: '0.4.118.179',
        ipRange: 41,
      })
    ).toBe('IP/Ranges');

    expect(
      helper.mapWhitelistType({
        ...common,
        userId: '1234',
      })
    ).toBe('Player UserID');

    expect(
      helper.mapWhitelistType({
        ...common,
        gamerTag: 'Player A',
      })
    ).toBe('Player GamerTag');

    expect(
      helper.mapWhitelistType({
        ...common,
        consoleId: 'ABCD-1234',
      })
    ).toBe('ConsoleID');
  });
});
