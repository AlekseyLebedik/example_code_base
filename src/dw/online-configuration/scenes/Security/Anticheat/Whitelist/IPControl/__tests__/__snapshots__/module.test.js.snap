// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security - Anticheat - Whitelist - IPControl Action Creators addIPNote action 1`] = `
Object {
  "type": "security/IP_CONTROL_NOTE_ADD",
  "values": "test",
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Action Creators deleteIPNote action 1`] = `
Object {
  "type": "security/IP_CONTROL_NOTE_DELETE",
  "values": "test",
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Action Creators fetchIPGroups action 1`] = `
Object {
  "append": "test",
  "params": "test",
  "type": "security/IP_CONTROL_FETCH_GROUPS",
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Action Creators fetchIPGroupsSuccess action 1`] = `
Object {
  "append": "test",
  "payload": "test",
  "type": "security/IP_CONTROL_FETCH_GROUPS_SUCCESS",
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Action Creators fetchIPNotes action 1`] = `
Object {
  "append": "test",
  "params": "test",
  "type": "security/IP_CONTROL_FETCH_NOTES",
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Action Creators fetchIPNotesSuccess action 1`] = `
Object {
  "append": "test",
  "payload": "test",
  "type": "security/IP_CONTROL_FETCH_NOTES_SUCCESS",
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Action Creators updateIPNote action 1`] = `
Object {
  "type": "security/IP_CONTROL_NOTE_UPDATE",
  "values": "test",
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer Test append = false 1`] = `
Object {
  "ipControl": Array [
    Object {
      "ipInt": 3,
    },
    Object {
      "ipInt": 4,
    },
  ],
  "nextPageToken": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer Test append = true 1`] = `
Object {
  "ipControl": Array [
    Object {
      "ipInt": 1,
    },
    Object {
      "ipInt": 2,
    },
    Object {
      "ipInt": 3,
    },
    Object {
      "ipInt": 4,
    },
  ],
  "nextPageToken": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer handle IP_CONTROL_ADD 1`] = `
Object {
  "addModalSubmitting": true,
  "addModalVisible": false,
  "ipControl": null,
  "nextPageToken": null,
  "q": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer handle IP_CONTROL_ADD_FAILED 1`] = `
Object {
  "addModalSubmitting": false,
  "addModalVisible": false,
  "ipControl": null,
  "nextPageToken": null,
  "q": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer handle IP_CONTROL_ADD_MODAL_CLOSE 1`] = `
Object {
  "addModalSubmitting": false,
  "addModalVisible": false,
  "ipControl": null,
  "nextPageToken": null,
  "q": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer handle IP_CONTROL_ADD_MODAL_OPEN 1`] = `
Object {
  "addModalSubmitting": false,
  "addModalVisible": true,
  "ipControl": null,
  "nextPageToken": null,
  "q": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer handle IP_CONTROL_ADD_SUCCESS 1`] = `
Object {
  "addModalSubmitting": false,
  "addModalVisible": false,
  "ipControl": Array [
    Object {
      "ipAddr": "127.0.0.1",
      "ipInt": 2130706433,
      "ipRange": 5,
    },
  ],
  "nextPageToken": null,
  "q": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer handle IP_CONTROL_DELETE_SUCCESS 1`] = `
Object {
  "addModalSubmitting": false,
  "addModalVisible": false,
  "ipControl": Array [
    Object {
      "creatorNote": "",
      "group": "Unassigned Group",
      "groupID": 0,
      "ipAddr": "0.80.164.230",
      "ipInt": 5285094,
      "ipRange": 32,
      "note": "Click to add note",
      "type": "IP/Ranges",
      "updatedAt": "N/A",
    },
  ],
  "nextPageToken": "CAE",
  "q": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer handle IP_CONTROL_FETCH_SUCCESS: empty results 1`] = `
Object {
  "addModalSubmitting": false,
  "addModalVisible": false,
  "ipControl": Array [],
  "nextPageToken": undefined,
  "q": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer handle IP_CONTROL_FETCH_SUCCESS: ipControl and nextPagetoken assignation 1`] = `
Object {
  "addModalSubmitting": false,
  "addModalVisible": false,
  "ipControl": Array [
    Object {
      "creatorNote": "",
      "group": "Unassigned Group",
      "groupID": 0,
      "ipAddr": "0.4.118.179",
      "ipInt": 292531,
      "ipRange": 41,
      "note": "Click to add note",
      "type": "IP/Ranges",
      "updatedAt": "N/A",
    },
    Object {
      "creatorNote": "",
      "group": "Unassigned Group",
      "groupID": 0,
      "ipAddr": "0.80.164.230",
      "ipInt": 5285094,
      "ipRange": 32,
      "note": "Click to add note",
      "type": "IP/Ranges",
      "updatedAt": "N/A",
    },
  ],
  "nextPageToken": "CAE",
  "q": undefined,
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer handle IP_CONTROL_SEARCH 1`] = `
Object {
  "addModalSubmitting": false,
  "addModalVisible": false,
  "ipControl": null,
  "nextPageToken": null,
  "q": "blah",
}
`;

exports[`Security - Anticheat - Whitelist - IPControl Reducer return the initial state 1`] = `
Object {
  "addModalSubmitting": false,
  "addModalVisible": false,
  "ipControl": null,
  "nextPageToken": null,
  "q": undefined,
}
`;
